haas/cli.py

```diff
@@ -79,17 +79,38 @@ def object_url(*args):
         url += '/' + urllib.quote(arg,'')
     return url
 
+def do_request(fn, url, data={}):
+    """Helper function for making HTTP requests against the API.
+
+    Arguments:
+
+        `fn` - a function from the requests library, one of requests.put,
+               requests.get...
+        `url` - The url to make the request to
+        `data` - the body of the request.
+
+    If the environment variables HAAS_USERNAME and HAAS_PASSWORD are
+    defined, The request will use HTTP basic auth to authenticate, with
+    the given username and password.
+    """
+    kwargs = {}
+    username = os.getenv('HAAS_USERNAME')
+    password = os.getenv('HAAS_PASSWORD')
```

Comment:
I think it'd be ideal to support CLI parameters for this, but implementing argparse (or something equivalent) would be out of scope for this PR and is part of #431.